### PR TITLE
Use if-let chains

### PR DESCRIPTION
**Pull request by crawfordleeds**
Per Issue #195, I have gone through and added some if-let chains instead of pattern matching. Would appreciate any feedback you may have. Thanks!

### CODE REVIEW

File: src/producer.rs [hunk 1]
```diff
@@ -140,10 +140,13 @@ impl Archive {
     fn check_file(file: FilePath, checker: &Fn(&mut Read) -> bool) -> bool {
         match file {
             FilePath::File(reader) => checker(reader),
-            FilePath::Path(path) => match File::open(path) {
-                Ok(mut f) => checker(&mut f),
-                Err(_) => false,
-            },
+            FilePath::Path(path) => {
```
Comment (marco-c) [inline src/producer.rs]: This one in particular was probably a bit more readable with the match instead of if-let

File: src/producer.rs [hunk 2]
```diff
@@ -250,23 +253,31 @@ impl Archive {
     pub fn extract(&self, name: &str, path: &PathBuf) -> bool {
         let dest_parent = path.parent().unwrap();
         if !dest_parent.exists() {
-            match fs::create_dir_all(dest_parent) {
-                Ok(_) => {}
-                Err(_) => panic!("Cannot create parent directory"),
-            };
+            if let Ok(_) = fs::create_dir_all(dest_parent) {}
```
Comment (marco-c) [inline src/producer.rs]: Maybe here you could do:
```Rust
if let Err(_) = fs::create_dir_all(dest_parent) {
  panic!("Cannot create parent directory")
}
```
Or even:
```Rust
fs::create_dir_all(dest_parent).expect("Cannot create parent directory");
```

File: src/producer.rs [hunk 3]
```diff
@@ -250,23 +253,31 @@ impl Archive {
...
                 let mut zip = zip.borrow_mut();
                 let zipfile = zip.by_name(&name);
-                match zipfile {
-                    Ok(mut f) => {
-                        let mut file = File::create(&path).expect("Failed to create file");
-                        io::copy(&mut f, &mut file).expect("Failed to copy file from ZIP");
-                        true
-                    }
-                    Err(_) => false,
+                // match zipfile {
```
Comment (marco-c) [inline src/producer.rs]: Here you can remove the commented-out part.

**Comment by marco-c** [comment]
Thanks, it looks good overall!

File: src/producer.rs (hunk 3 above)
Comment (crawfordleeds) [reply src/producer.rs]: Oops, thanks :)

File: src/producer.rs [hunk 4]
```diff
@@ -143,7 +143,7 @@ impl Archive {
             FilePath::Path(path) => match File::open(path) {
                 Ok(mut f) => checker(&mut f),
                 Err(_) => false,
-            },
+            }
```
Comment (marco-c) [inline src/producer.rs]: Nit: could you bring back the comma to avoid affecting 'blame' information

Comment (crawfordleeds) [reply src/producer.rs]: Of course, thanks for your patience.

Comment (marco-c) [reply src/producer.rs]: No worries, that's really minor!

**Comment by crawfordleeds** [comment]
Any thoughts on why the codecov was so shot by these changes? I didn't really look into the tests, should I dig into that a bit?

**Comment by marco-c** [comment]
> Any thoughts on why the codecov was so shot by these changes? I didn't really look into the tests, should I dig into that a bit?

No, no worries, I think it's just imprecise. We have to fix it somehow.

**Comment by crawfordleeds** [comment]
> > Any thoughts on why the codecov was so shot by these changes? I didn't really look into the tests, should I dig into that a bit?
> 
> No, no worries, I think it's just imprecise. We have to fix it somehow.

Great, thanks!

**Pull request merged**